### PR TITLE
Disable Download All button and show unavailability message if batch zip file is not ready AB#12935

### PR DESCRIPTION
**Pull request by prasad14878**

### CODE REVIEW

File: Tests/UnitTests/fss-search-results.component.spec.ts [hunk 1]
```diff
@@ -175,11 +175,12 @@ describe('FssSearchResultsComponent', () => {
         let css = "[id='" + item.BatchID.value + "']";
         const home = fixture.debugElement.query(By.css(css)).nativeElement;
         if (item.allFilesZipSize) {
-          expect(home.querySelector("ukho-button").textContent).toEqual("Download all");
+          expect(home.querySelector("a").textContent).toEqual("Download all");
```
Comment (crdandoukho) [inline Tests/UnitTests/fss-search-results.component.spec.ts:178]: Please add an expectation to confirm the 'Download all' link is enabled.
Please add an expectation to confirm the 'note' (ukho-dialogue) element is not visible

File: Tests/UnitTests/fss-search-results.component.spec.ts [hunk 2]
```diff
@@ -175,11 +175,12 @@ describe('FssSearchResultsComponent', () => {
...
         }
         else
         {
-          expect(home.querySelector("ukho-button")).toBeNull();
+          expect(home.querySelector("ukho-dialogue").textContent).toEqual("'Download all' function will be available when the files have been prepared You can select and download individual files, or try again later ");
```
Comment (crdandoukho) [inline Tests/UnitTests/fss-search-results.component.spec.ts:182]: Please add an expectation to confirm the 'Download all' link is disabled.

File: Tests/UnitTests/fss-search-results.component.spec.ts (hunk 1 above)
Comment (crdandoukho) [inline Tests/UnitTests/fss-search-results.component.spec.ts:178]: Could you please add an expectation to prove the `batches` collection is not empty - to prove at least one iteration of the forEach loop will be executed

File: src/app/features/fss-search/fss-search-results/fss-search-results.component.ts [hunk 3]
```diff
@@ -108,6 +108,10 @@ export class FssSearchResultsComponent implements OnChanges {
       }
     }
   }
+
+  DownloadAll(){
```
Comment (crdandoukho) [inline src/app/features/fss-search/fss-search-results/fss-search-results.component.ts:112]: Please change to camel-case `downloadAll()` to match other identifiers

**Review by crdandoukho — CHANGES_REQUESTED** [review]

File: src/app/features/fss-search/fss-search-results/fss-search-results.component.ts (hunk 3 above)
Comment (prasad14878) [reply src/app/features/fss-search/fss-search-results/fss-search-results.component.ts]: Modified as per review comment

File: Tests/UnitTests/fss-search-results.component.spec.ts (hunk 1 above)
Comment (prasad14878) [reply Tests/UnitTests/fss-search-results.component.spec.ts]: Modified as per review comment

File: Tests/UnitTests/fss-search-results.component.spec.ts (hunk 2 above)
Comment (prasad14878) [reply Tests/UnitTests/fss-search-results.component.spec.ts]: Modified as per review comment

File: Tests/UnitTests/fss-search-results.component.spec.ts (hunk 1 above)
Comment (prasad14878) [reply Tests/UnitTests/fss-search-results.component.spec.ts]: Modified as per review comment

**Review by crdandoukho — APPROVED** [review]
Thanks Prasad for your changes

Looks good

**Review by bhalcha13813 — APPROVED** [review]
LGTM

**Comment by crdandoukho** [comment]
Changes in this branch included in merge of 12858

**Pull request closed without merging**